src/Umbraco.Cms.Integrations.Commerce.Shopify/Services/ShopifyService.cs

```diff
@@ -106,13 +106,25 @@ public EditorSettings GetApiConfiguration()
             // validate OAuth configuration if AuthorizationService is used.
             // if authorization is managed through UmbracoAuthorizationService, the properties client ID and proxy URL are set correctly.
             if (_settings.UseUmbracoAuthorization)
-                return new EditorSettings { IsValid = true, Type = ConfigurationType.OAuth };
+            {
+                var editorSettings = new EditorSettings { IsValid = true, Type = ConfigurationType.OAuth };
+
+                editorSettings.IsConnected = ValidateAccessToken().ConfigureAwait(false).GetAwaiter().GetResult().IsValid;
```

Comment:
Could this method be made `async` such that you can avoid the `.GetAwaiter().GetResult()`?

Reply:
Updated and corrected the additional references.